virtcontainers/clh_test.go

```diff
@@ -0,0 +1,718 @@
+		src: defaultSource,
+	})
+	cli, err := director.Build(builder)
+	assert.NotNil(cli.args)
+	assert.Nil(err)
+	assert.Equal(cliToString(cli.args), "--rng src="+defaultSource+",iommu=off")
+
+	builder.SetRng(&CLIRng{
+		src:   defaultSource,
+		iommu: true,
+	})
+	cli, err = director.Build(builder)
+	assert.NotNil(cli.args)
+	assert.Nil(err)
+	assert.Equal(cliToString(cli.args), "--rng src="+defaultSource+",iommu=on")
+}
+
+//
+// --vsock <vsock> Virtio VSOCK parameters "cid=<context_id>,sock=<socket_path>,iommu=on|off"
+//
+func TestClhCliVsock(t *testing.T) {
+
+	assert := assert.New(t)
+	builder := &DefaultCLIBuilder{}
+	director := &CommandLineDirector{}
+	defaultSocket := "/a/b/c.sock"
+	defaultCid := uint32(12345)
+
+	builder.SetVsock(&CLIVsock{
+		cid:        defaultCid,
+		socketPath: defaultSocket,
+	})
+	cli, err := director.Build(builder)
+	assert.NotNil(cli.args)
+	assert.Nil(err)
+	assert.Equal(cliToString(cli.args),
+		"--vsock cid="+strconv.FormatUint(uint64(defaultCid), 10)+
+			",sock="+defaultSocket+
+			",iommu=off")
+
+	builder.SetVsock(&CLIVsock{
+		cid:        defaultCid,
+		socketPath: defaultSocket,
+		iommu:      true,
+	})
+	cli, err = director.Build(builder)
+	assert.NotNil(cli.args)
+	assert.Nil(err)
+	assert.Equal(cliToString(cli.args),
+		"--vsock cid="+strconv.FormatUint(uint64(defaultCid), 10)+
+			",sock="+defaultSocket+
+			",iommu=on")
+
+}
+
+func TestClhCreateSandbox(t *testing.T) {
+	clhConfig := newClhConfig()
+	clh := &cloudHypervisor{
+		config: clhConfig,
+	}
+	assert := assert.New(t)
+
+	sandbox := &Sandbox{
+		ctx: context.Background(),
+		id:  "testSandbox",
+		config: &SandboxConfig{
+			HypervisorConfig: clhConfig,
+		},
+	}
+
+	vcStore, err := store.NewVCSandboxStore(sandbox.ctx, sandbox.id)
+	assert.NoError(err)
+
+	sandbox.store = vcStore
+
+	// Create the hypervisor fake binary
+	testClhPath := filepath.Join(testDir, testHypervisor)
+	_, err = os.Create(testClhPath)
+	assert.NoError(err)
+
+	// Create parent dir path for hypervisor.json
+	parentDir := store.SandboxConfigurationRootPath(sandbox.id)
+	assert.NoError(os.MkdirAll(parentDir, store.DirMode))
+
+	err = clh.createSandbox(context.Background(), sandbox.id, NetworkNamespace{}, &sandbox.config.HypervisorConfig, sandbox.store, false)
+	assert.NoError(err)
+	assert.NoError(os.RemoveAll(parentDir))
+	assert.Exactly(clhConfig, clh.config)
+}
+
+func TestClhAddDeviceNet(t *testing.T) {
+
+	defaultEndpointName := "tap1"
+	defaultMac := "55:44:33:22:11:00"
+	assert := assert.New(t)
+	clh := &cloudHypervisor{
+		ctx:        context.Background(),
+		cliBuilder: &DefaultCLIBuilder{},
+	}
+
+	tep, _ := createTapNetworkEndpoint(0, defaultEndpointName)
+	tep.TapInterface.TAPIface.HardAddr = defaultMac
+
+	err := clh.addDevice(Endpoint(tep), netDev)
+	assert.NoError(err)
+
+	director := &CommandLineDirector{}
+
+	cli, err := director.Build(clh.cliBuilder)
+	assert.NoError(err)
+	assert.NotNil(cli)
+
+	netarg, errr := getCliOption(cli.args, "--net")
```

Comment:
Nit: should this be `err`? (same comment for other functions below).

Reply:
changed.